src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.ts

```diff
@@ -202,6 +202,38 @@ export class ERC721Standard {
     });
   };
 
+  getTokenURIAndImage = async (
+    address: string,
+    ipfsGateway: string,
+    tokenId: string,
+  ) => {
+    let tokenURI, image;
+
+    try {
+      tokenURI = await this.getTokenURI(address, tokenId);
+      if (tokenURI.startsWith('ipfs://')) {
+        tokenURI = getFormattedIpfsUrl(ipfsGateway, tokenURI, true);
+      }
+    } catch {
+      // ignore
+    }
+
+    if (tokenURI) {
+      try {
+        const response = await timeoutFetch(tokenURI);
```

Comment:
This may potentially leak users IP. Needs to be an issue created if there isn't already.